### DISCUSSION TITLE
Reusing Users in a custom element overview?

### DISCUSSION THREAD

**Discussion by janhenckens** [craftcms/cms#13215] [Q&A] [answered]
We need to build an extensive member management system for an upcoming project, where users need a login to the site to view protected content, but whose user account also needs to be an object in a mini-crm system where our client can keep track of memberships, payments , etc.

My initial idea was to create a new element, `Member`, that extends Craft's `User` element. That way I can give it its own index page in the CP, custom sources based on fields & categories, custom actions,  custom detail URLs, etc.

But I was wondering if extending User is a good idea, and if there is a way to make a separate overview from the existing element?

Thanks for your input on this, I hope my explanation is somewhat clear :)

**Comment by brandonkelly** [comment, marked as answer]
I would make `Member` its own element, with a 1-to-1 mapping between members and users. You can trigger a `Member` create after `User` save, or vise-versa, and give `Member` a `$userId` property/foreign key that points back to its proxied `User` element.

**Reply by janhenckens** [reply]
Thanks, that's what we had in mind as well 👍
